brainglobe_stitch/image_mosaic.py

```diff
@@ -261,3 +264,88 @@ def write_big_stitcher_tile_config(self, meta_file_name: Path) -> None:
                 )
 
         return
+
+    def stitch(
+        self,
+        fiji_path: Path,
+        resolution_level: int,
+        selected_channel: str,
+    ) -> None:
+        """
+        Stitch the tiles in the image using BigStitcher.
+
+        Parameters
+        ----------
+        fiji_path: Path
+            The path to the Fiji application.
+        resolution_level: int
+            The resolution level to stitch the tiles at.
+        selected_channel: str
+            The name of the channel to stitch.
+        """
+
+        # If selected_channel is an empty string then stitch based on
+        # all channels
+        all_channels = len(selected_channel) == 0
+        channel_int = -1
+
+        # Extract the wavelength from the channel name
+        if not all_channels:
+            try:
+                channel_int = int(selected_channel.split()[0])
+            except ValueError:
+                print("Invalid channel name.")
+                raise
+
+        # Extract the downsample factors for the selected resolution level
+        downsample_z, downsample_y, downsample_x = self.tiles[
+            0
+        ].resolution_pyramid[resolution_level]
+
+        assert self.xml_path is not None
+        assert self.tile_config_path is not None
+
+        result = run_big_stitcher(
+            fiji_path,
+            self.xml_path,
+            self.tile_config_path,
+            all_channels,
+            channel_int,
+            downsample_x=downsample_x,
+            downsample_y=downsample_y,
+            downsample_z=downsample_z,
+        )
+
+        big_stitcher_output_path = self.directory / "big_stitcher_output.txt"
+
+        with open(big_stitcher_output_path, "w") as f:
+            f.write(result.stdout)
+            f.write(result.stderr)
+
+        # Print the output of BigStitcher to the command line
+        print(result.stdout)
+
+        # Wait for the BigStitcher to write XML file
+        # Need to find a better way to do this
+        sleep(1)
+
+        self.read_big_stitcher_transforms()
+
+    def read_big_stitcher_transforms(self) -> None:
+        """
+        Read the BigStitcher transforms from the XML file and update the tile
+        positions accordingly.
+        """
+        z_size, y_size, x_size = self.tiles[0].data_pyramid[0].shape
+        assert self.xml_path is not None
+        stitched_translations = get_big_stitcher_transforms(
+            self.xml_path, z_size, y_size, x_size
```

Comment:
That's actually a bug! Thankfully, I wasn't actually using the "end" outputs from this functions just the "start" so the incorrect sizes weren't having any effect. I've simplified this function now so that it only returns the relevant information.